src/Analysis/Ast/Impl/Analyzer/PythonAnalyzer.cs

```diff
@@ -196,18 +196,16 @@ public Task WaitForCompleteAnalysisAsync(CancellationToken cancellationToken = d
             return optionsProvider?.Options?.LintingEnabled == false ? Array.Empty<DiagnosticsEntry>() : result;
         }
 
-        public async Task ResetAnalyzer(bool full) {
-            if (full) {
-                var interpreter = _services.GetService<IPythonInterpreter>();
-                var builtins = interpreter.ModuleResolution.BuiltinsModule;
-                builtins.SetAst(builtins.Analysis.Ast);
-
-                await interpreter.TypeshedResolution.ReloadAsync();
-                await interpreter.ModuleResolution.ReloadAsync();
-            }
+        public async Task ResetAnalyzer() {
+            var interpreter = _services.GetService<IPythonInterpreter>();
+            var builtins = interpreter.ModuleResolution.BuiltinsModule;
+            builtins.SetAst(builtins.Analysis.Ast);
+
+            await interpreter.TypeshedResolution.ReloadAsync();
+            await interpreter.ModuleResolution.ReloadAsync();
 
             lock (_syncObj) {
-                _forceGCOnNextSession = _forceGCOnNextSession || full;
+                _forceGCOnNextSession = true;
```

Comment:
If we reload `TypeshedResolution` and `BuiltinsModule`, there is no reason for a split bleow. _analysisEntries should be completely cleared, `_dependencyResolver` can be thrown away.

Reply:
Sort of; the way we create the builtins module is really contrived and currently can only be created once when the interpreter object is created. What I had to do here is only a partial fix; we don't have a way to do a complete reload including the builtin module, so I kept the existing code and allowed it to remove more.

I'd like for this to be fixed in the future so we can have a "real" reload that drops everything but file contents and does a complete restart, but I think that'd be a bigger change.